core/flow/rule_manager.go

```diff
@@ -224,27 +230,106 @@ func onRuleUpdate(rules []*Rule) (err error) {
 
 	tcMux.Lock()
 	tcMap = m
-	currentRules = rules
 	tcMux.Unlock()
+	currentRules = rawResRulesMap
 
 	logging.Debug("[Flow onRuleUpdate] Time statistic(ns) for updating flow rule", "timeCost", util.CurrentTimeNano()-start)
-	logRuleUpdate(resRulesMap)
+	logRuleUpdate(validResRulesMap)
 	return nil
 }
 
 // LoadRules loads the given flow rules to the rule manager, while all previous rules will be replaced.
 // the first returned value indicates whether do real load operation, if the rules is the same with previous rules, return false
 func LoadRules(rules []*Rule) (bool, error) {
-	// TODO: rethink the design
+	resRulesMap := make(map[string][]*Rule, 9)
```

Comment:
`9` may be a little strange for people who aren't familiar with the implementation details. How about 16?

Reply:
ok